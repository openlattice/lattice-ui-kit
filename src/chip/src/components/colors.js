import * as Colors from '../../../colors';

const {
  NEUTRAL,
  PURPLE,
  BLUE,
  TEAL,
  GREEN,
  YELLOW,
  ORANGE,
  RED,
  MAGENTA,
  VIOLET
} = Colors;

export default {
  default: {
    background: NEUTRAL.N00,
    color: NEUTRAL.N600
  },
  blue: {
    background: BLUE.B00,
    color: BLUE.B400,
    clickable: BLUE.B100
  },
  green: {
    background: GREEN.G00,
    color: GREEN.G400,
    clickable: GREEN.G100
  },
  magenta: {
    background: MAGENTA.M00,
    color: MAGENTA.M400,
    clickable: MAGENTA.M100
  },
  orange: {
    background: ORANGE.O00,
    color: ORANGE.O500,
    clickable: ORANGE.O100
  },
  purple: {
    background: PURPLE.P00,
    color: PURPLE.P400,
    clickable: PURPLE.P100
  },
  red: {
    background: RED.R00,
    color: RED.R400,
    clickable: RED.R100
  },
  teal: {
    background: TEAL.T00,
    color: TEAL.T500,
    clickable: TEAL.T100
  },
  violet: {
    background: VIOLET.V00,
    color: VIOLET.V400,
    clickable: VIOLET.V100
  },
  yellow: {
    background: YELLOW.Y00,
    color: YELLOW.Y500,
    clickable: YELLOW.Y100
  },
  primary: {
    color: 'white'
  },
  secondary: {
    color: PURPLE.P400
  },
};
