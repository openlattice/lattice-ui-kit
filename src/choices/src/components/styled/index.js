// @ flow
import CheckboxIndicator from './CheckboxIndicator';
import ChoiceLabel from './ChoiceLabel';
import RadioIndicator from './RadioIndicator';
import { CheckboxInput, RadioInput } from './ChoiceInputs';

export {
  CheckboxIndicator,
  CheckboxInput,
  ChoiceLabel,
  RadioIndicator,
  RadioInput
};
