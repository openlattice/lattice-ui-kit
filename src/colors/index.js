/*
 * @flow
 */

/*
 * Inspiration:
 * https://atlassian.design/guidelines/brand/color-1
 * https://bitbucket.org/atlassian/atlaskit-mk-2/src/master/packages/core/theme
 * https://evergreen.surge.sh/components/colors
 */

import {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
} from './src/Colors';

export {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
};

export default {
  BLUE,
  GREEN,
  MAGENTA,
  NEUTRAL,
  ORANGE,
  PURPLE,
  RED,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
};
