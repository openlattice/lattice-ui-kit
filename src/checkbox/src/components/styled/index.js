// @flow
import styled, { css } from 'styled-components';
import { NEUTRALS, PURPLES } from '../../../../colors';

const CheckboxLabel = styled.label`
  align-items: center;
  color: ${NEUTRALS[0]};
  display: inline-flex;
  font-size: 14px;
  font-weight: normal;
  margin: 10px 0;
  min-height: 20px;
  padding: 0 10px 0 30px;
  pointer-events: ${props => (props.readOnly ? 'none' : 'auto')};
  position: relative;
`;


type IndicatorProps = {
  checked :boolean;
};

const getAfterStyles = ({ checked } :IndicatorProps) => {
  if (!checked) {
    return css`
      display: none;
    `;
  }

  return css`
    content: '';
    position: absolute;
    left: 8px;
    top: 4px;
    width: 3px;
    height: 8px;
    border: solid white;
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
  `;
};

const CheckboxIndicator = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  background: ${PURPLES[6]};
  border-radius: 3px;

  :after {
    ${getAfterStyles}
    }
  }
`;

const CheckboxInput = styled.input.attrs(() => ({
  type: 'checkbox'
}))`
  position: absolute;
  z-index: -1;
  opacity: 0;
  vertical-align: middle;
  
  :hover + ${CheckboxIndicator},
  :focus + ${CheckboxIndicator} {
    background: ${NEUTRALS[1]};
  }

  :checked + ${CheckboxIndicator} {
    background: ${PURPLES[2]};
  }

  :checked:hover + ${CheckboxIndicator},
  :checked:focus + ${CheckboxIndicator} {
    background: ${PURPLES[1]};
  }

  :disabled + ${CheckboxIndicator} {
    background: ${PURPLES[6]};
    cursor: not-allowed;
  }

  :checked:disabled + ${CheckboxIndicator} {
    background: ${NEUTRALS[2]};
  }
`;


export {
  CheckboxIndicator,
  CheckboxInput,
  CheckboxLabel,
};
