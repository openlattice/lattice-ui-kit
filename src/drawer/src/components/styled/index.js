import styled, { css } from 'styled-components';

const getPosition = (props) => {
  const { side } = props;
  switch (side) {
    case 'right':
      return css`
        right: 0;
      `;

    case 'left':
    default:
      return css`
        left: 0;
      `;
  }
};

const DrawerWrapper = styled.div`
  position: fixed;
  background-color: white;
  height: 100%;
  width: 300px;
  top: 0;
  z-index: 900;
  box-shadow: 0 0 30px 0 rgba(0, 0, 0, 0.2);

  ${getPosition};
`;

export {
  // eslint-disable-next-line import/prefer-default-export
  DrawerWrapper
};
