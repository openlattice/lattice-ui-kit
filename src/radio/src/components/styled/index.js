// @flow
import styled, { css } from 'styled-components';
import { NEUTRALS, PURPLES, WHITE } from '../../../../colors';

const RadioLabel = styled.label`
  align-items: center;
  color: ${NEUTRALS[0]};
  display: inline-flex;
  margin: 10px 0;
  min-height: 20px;
  padding-left: 30px;
  pointer-events: ${props => (props.readOnly ? 'none' : 'auto')};
  position: relative;
`;

type IndicatorProps = {
  checked :boolean;
};

const getAfterStyles = ({ checked } :IndicatorProps) => {
  if (!checked) {
    return css`
      display: none;
    `;
  }

  return css`
    content: '';
    background-color: ${WHITE};
    border-radius: 50%;
    height: 8px;
    left: 6px;
    position: absolute;
    top: 6px;
    width: 8px;
  `;
};

const RadioIndicator = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  background: ${PURPLES[6]};
  border-radius: 50%;

  :after {
    ${getAfterStyles}
  }
`;

const RadioInput = styled.input.attrs(() => ({
  type: 'radio'
}))`
  position: absolute;
  z-index: -1;
  opacity: 0;
  vertical-align: middle;

  :hover + div,
  :focus + div {
    background: ${NEUTRALS[1]};
  }

  :checked + div {
    background: ${PURPLES[2]};
  }

  :checked:hover + div,
  :checked:focus + div {
    background: ${PURPLES[1]};
  }

  :disabled + div {
    background: ${PURPLES[6]};
    cursor: not-allowed;
  }

  :checked:disabled + div {
    background: ${NEUTRALS[2]};
  }
`;

export {
  RadioLabel,
  RadioInput,
  RadioIndicator,
};
