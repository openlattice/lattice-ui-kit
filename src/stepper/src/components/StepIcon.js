// @flow
import React from 'react';

import { faCheck } from '@fortawesome/pro-regular-svg-icons';
import { faCircle } from '@fortawesome/pro-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { IconLayer, StepIndex } from './styled';

import {
  NEUTRALS,
  PURPLE,
  WHITE
} from '../../../colors';

type Props = {
  active :boolean;
  className ? :string;
  complete :boolean;
  index :number;
};

const StepIcon = ({
  active,
  className,
  complete,
  index
} :Props) => {
  const styledClassName = className ? `${className} fa-layers` : 'fa-layers';
  return (
    <IconLayer className={styledClassName}>
      <FontAwesomeIcon icon={faCircle} color={(active || complete) ? PURPLE.P300 : NEUTRALS[1]} />
      {
        complete
          ? <FontAwesomeIcon icon={faCheck} color={WHITE} transform="shrink-7" />
          : <StepIndex className="fa-layers-text fa-inverse">{index}</StepIndex>
      }
    </IconLayer>
  );
};

StepIcon.defaultProps = {
  className: undefined
};

export default StepIcon;
