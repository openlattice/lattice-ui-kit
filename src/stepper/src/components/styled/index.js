import styled, { css } from 'styled-components';
import { NEUTRALS } from '../../../../colors';

const IconLayer = styled.span`
  font-size: 24px;
  margin-right: 8px;
`;

const getDividerOrientationStyles = ({ vertical }) => {
  if (vertical) {
    return css`
      min-height: 20px;
      border-left: 2px solid ${NEUTRALS[4]};
       /* step padding + half icon - 1 */
      margin-left: 19px;
    `;
  }

  return css`
    border-top: 2px solid ${NEUTRALS[4]};
  `;
};

const StepDivider = styled.div`
  flex: 1 1 auto;
  ${getDividerOrientationStyles};
`;

const StepIndex = styled.span`
  font-size: 14px;
  font-weight: bold;
`;

const StepWrapper = styled.div`
  display: flex;
  flex: 0 1 auto;
  align-items: center;
  padding: 10px;
  cursor: ${props => props.onClick && 'pointer'};
`;

const getStepperOrientationStyles = ({ vertical }) => {
  if (vertical) {
    return css`
      align-items: flex-start;
      flex-direction: column;
      justify-content: start;
    `;
  }

  return css`
    flex-direction: row;
    justify-content: space-around;
  `;
};

const StepperWrapper = styled.div`
  align-items: center;
  display: flex;
  flex: 1 0 auto;
  ${getStepperOrientationStyles};
`;

export {
  IconLayer,
  StepDivider,
  StepIndex,
  StepperWrapper,
  StepWrapper,
};
