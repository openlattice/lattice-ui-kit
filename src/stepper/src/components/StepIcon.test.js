import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheck } from '@fortawesome/pro-regular-svg-icons';

import StepIcon from './StepIcon';
import { PURPLES } from '../../../colors';
import { StepIndex } from './styled';

describe('StepIcon', () => {

  describe('default', () => {
    test('should match snapshot', () => {
      const wrapper = mount(<StepIndex index={0} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('active should set color', () => {
      const wrapper = shallow(<StepIcon index={0} active />);
      expect(wrapper.find(FontAwesomeIcon).prop('color')).toBe(PURPLES[2]);
    });

    test('render index + 1', () => {
      const wrapper = shallow(<StepIcon index={0} />);
      expect(wrapper.find(StepIndex).text()).toBe('0');
    });
  });

  describe('complete', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = mount(<StepIcon index={0} complete />);
    });

    test('should match snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('complete should set color', () => {
      expect(wrapper.find(FontAwesomeIcon).first().prop('color')).toBe(PURPLES[2]);
    });

    test('should should render icon instead of index', () => {
      expect(wrapper.find(StepIndex)).toHaveLength(0);
      expect(wrapper.find(FontAwesomeIcon).last().prop('icon')).toBe(faCheck);
    });

  });


});
