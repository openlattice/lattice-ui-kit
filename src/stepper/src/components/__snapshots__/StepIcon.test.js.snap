// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepIcon complete should match snapshot 1`] = `
.c0 {
  font-size: 24px;
  min-width: 24px;
}

<StepIcon
  complete={true}
  index={0}
>
  <styled__IconLayer
    className="fa-layers"
  >
    <StyledComponent
      className="fa-layers"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__IconLayer-g8vnjr-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "font-size:24px;min-width:24px;",
            ],
          },
          "displayName": "styled__IconLayer",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__IconLayer-g8vnjr-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <span
        className="c0 fa-layers"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          color="#8045ff"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f111",
                "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z",
              ],
              "iconName": "circle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-circle fa-w-16 "
            color="#8045ff"
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
        <FontAwesomeIcon
          border={false}
          className=""
          color="#ffffff"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f00c",
                "M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z",
              ],
              "iconName": "check",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform="shrink-7"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-check fa-w-16 "
            color="#ffffff"
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={
              Object {
                "transformOrigin": "0.5em 0.5em",
              }
            }
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              style={Object {}}
              transform="translate(256 256)"
            >
              <g
                style={Object {}}
                transform="translate(0, 0)  scale(0.5625, 0.5625)  rotate(0 0 0)"
              >
                <path
                  d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                  fill="currentColor"
                  style={Object {}}
                  transform="translate(-256 -256)"
                />
              </g>
            </g>
          </svg>
        </FontAwesomeIcon>
      </span>
    </StyledComponent>
  </styled__IconLayer>
</StepIcon>
`;

exports[`StepIcon default should match snapshot 1`] = `
.c0 {
  font-size: 14px;
  font-weight: bold;
}

<styled__StepIndex
  index={0}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "styled__StepIndex-g8vnjr-2",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "font-size:14px;font-weight:bold;",
          ],
        },
        "displayName": "styled__StepIndex",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "styled__StepIndex-g8vnjr-2",
        "target": "span",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    index={0}
  >
    <span
      className="c0"
    />
  </StyledComponent>
</styled__StepIndex>
`;
